pulp_rpm/app/depsolving.py

```diff
@@ -830,12 +810,246 @@ def run_solver_jobs(jobs):
+                "SOLVER_RULE_PKG_SELF_CONFLICT",
+                "The package conflicts with itself. This is not allowed by older rpm versions.",
+            )
+        elif rule == solv.Solver.SOLVER_RULE_PKG_CONFLICTS:
+            return (
+                "SOLVER_RULE_PKG_CONFLICTS",
+                "To fulfill the dependencies two packages need to be installed, but one of the packages contains a conflict with the other one.",
+            )
+        elif rule == solv.Solver.SOLVER_RULE_PKG_SAME_NAME:
+            return (
+                "SOLVER_RULE_PKG_SAME_NAME",
+                "The dependencies can only be fulfilled by multiple versions of a package, but installing multiple versions of the same package is not allowed.",
+            )
+        elif rule == solv.Solver.SOLVER_RULE_PKG_OBSOLETES:
+            return (
+                "SOLVER_RULE_PKG_OBSOLETES",
+                "To fulfill the dependencies two packages need to be installed, but one of the packages obsoletes the other one.",
+            )
+        elif rule == solv.Solver.SOLVER_RULE_PKG_IMPLICIT_OBSOLETES:
+            return (
+                "SOLVER_RULE_PKG_IMPLICIT_OBSOLETES",
+                "To fulfill the dependencies two packages need to be installed, but one of the packages has provides a dependency that is obsoleted by the other one. See the POOL_FLAG_IMPLICITOBSOLETEUSESPROVIDES flag.",
+            )
+        elif rule == solv.Solver.SOLVER_RULE_PKG_INSTALLED_OBSOLETES:
+            return (
+                "SOLVER_RULE_PKG_INSTALLED_OBSOLETES",
+                "To fulfill the dependencies a package needs to be installed that is obsoleted by an installed package. See the POOL_FLAG_NOINSTALLEDOBSOLETES flag.",
+            )
+        elif rule == solv.Solver.SOLVER_RULE_JOB_NOTHING_PROVIDES_DEP:
+            return (
+                "SOLVER_RULE_JOB_NOTHING_PROVIDES_DEP",
+                "The user asked for installation of a package providing a specific dependency, but no available package provides it.",
+            )
+        elif rule == solv.Solver.SOLVER_RULE_JOB_UNKNOWN_PACKAGE:
+            return (
+                "SOLVER_RULE_JOB_UNKNOWN_PACKAGE",
+                "The user asked for installation of a package with a specific name, but no available package has that name.",
+            )
+        elif rule == solv.Solver.SOLVER_RULE_JOB_PROVIDED_BY_SYSTEM:
+            return (
+                "SOLVER_RULE_JOB_PROVIDED_BY_SYSTEM",
+                "The user asked for the erasure of a dependency that is provided by the system (i.e. for special hardware or language dependencies), this cannot be done with a job.",
+            )
+        elif rule == solv.Solver.SOLVER_RULE_JOB_UNSUPPORTED:
+            return (
+                "SOLVER_RULE_JOB_UNSUPPORTED",
+                "The user asked for something that is not yet implemented, e.g. the installation of all packages at once.",
+            )
+
+    # TODO: group by repo?
```

Comment:
Yes please - knowing what repo is involved makes it easier for a user to understand what's going on. Spitting out the context prior to going through the results, and grouping results by repo, will make the output more immediately-useable when we're debugging.

Reply:
It already prints what repo the package is from so you'll always know that, I'm not sure if grouping it would provide much extra utility or not.

Reply:
Ah ok. Hm, do you have an example of a significant-amount-of-output you could throw up as a gist somewhere? Might help decide how useful (or not) it would be.

Reply:
Grouping was my original idea but the reason I thought it might not be so helpful is that ultimately if there's hundreds of packages listed it's going to be kind of hard to figure out what section of the file you're in.